### DISCUSSION TITLE
run a job when a nerwer image uploads to docker hub using dockerhub webhook

### DISCUSSION THREAD

**Discussion by harshakp06** [community/community#101824] [Actions]
### Select Topic Area

General

### Body

I am look for how to run github action job when a new image uploads to dockerhub repo 
we can implement with dockerhub webhook - but looking solution to connect dockerhub webhook with github actions

**Comment by ghost** [comment]
To run a GitHub Actions job when a new image is pushed to Docker Hub, you'll need to set up a Docker Hub webhook and a corresponding GitHub Actions workflow that can be triggered by this webhook. Here's how you can do it:

1. Create a GitHub Workflow with a Webhook Event:
- In your GitHub repository, create a new workflow file (e.g., .github/workflows/dockerhub-webhook.yml).
- Define a workflow that uses the repository_dispatch event, which can be triggered by an external webhook.
Here's an example of what your workflow file could look like:
```
name: Docker Hub Image Pushed

on:
  repository_dispatch:
    types: [dockerhub-push]

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - name: Checkout
      uses: actions/checkout@v2
    # Add additional steps to handle the Docker Hub image push event
```
2. Create a Webhook Endpoint:
- Set up an endpoint on your server or a serverless function (e.g., AWS Lambda, Azure Functions, or a GitHub Pages site) that listens for POST requests from Docker Hub.
- This endpoint should send a repository_dispatch event to the GitHub API when it receives a webhook from Docker Hub.

3. Configure the Server Endpoint to Trigger GitHub Actions:
- When your endpoint receives a webhook from Docker Hub, it should authenticate with GitHub and then send a repository_dispatch event to your repository. Here's an example of how to do this using curl:
```
curl -X POST \
  -H "Accept: application/vnd.github.v3+json" \
  -H "Authorization: token YOUR_GITHUB_TOKEN" \
  https://api.github.com/repos/YOUR_USERNAME/YOUR_REPO/dispatches \
  -d '{"event_type": "dockerhub-push"}'
```

4. Set up a Docker Hub Webhook:
- Go to your Docker Hub repository settings, find the Webhooks section, and add a new webhook.
- Set the webhook URL to your server or serverless function endpoint from step 2.

5. Test the Integration:
- Push a new image to Docker Hub and ensure that the webhook is sent.
- Check if your server or serverless function receives the webhook and sends the repository_dispatch event to GitHub.
- Verify that the GitHub Actions workflow is triggered.

**Remember to protect your server or serverless function with proper authentication mechanisms to prevent unauthorized use, and never expose your GitHub token.**
